ability to add no track

naazy opened the pull request:
ability to add no track #40

roryc89 commented on src/MainUpdate.elm:
```diff
@@ -331,7 +331,10 @@ update msg model =
             UpdateNewSessionTrack newTrackId ->
-                ( (updateNewSession model (\ns -> { ns | trackId = (toInt model newTrackId) })), Cmd.none )
+                if newTrackId == "" then
+                    ( (updateNewSession model (\ns -> { ns | trackId = Nothing })), Cmd.none )
```
I dont think we need double parens here or below.

Also we should try to use types that match their possible values.  If `newTrackId` can only be an integer or "null" then we should use an `Maybe Int` type

naazy replied:
Gotcha

roryc89 commented on src/NewSessionView.elm:
```diff
@@ -65,6 +65,9 @@ view context model =
                     (List.map (\c -> option [ value (toString c.id) ] [ text c.name ]) model.columns)
 
+        noTracksDropdownOption =
+            option [ value (""), selected (context.session.trackId == Nothing) ] [ text "No track" ]
```
unneeded parens here

roryc89 commented on src/NewSessionView.elm:
```diff
@@ -121,7 +124,7 @@ view context model =
                     , br [] []
                     , select [ id "track-input", onInput UpdateNewSessionTrack ]
-                        (List.map (\t -> option [ value (toString t.id), selected (context.session.trackId == t.id) ] [ text t.name ]) model.tracks)
+                        (noTracksDropdownOption :: List.map (\t -> option [ value (toString t.id), selected (context.session.trackId == Just (t.id)) ] [ text t.name ]) model.tracks)
```
unneeded parens around t.id here

roryc89 commented on src/TableView.elm:
```diff
@@ -314,9 +318,13 @@ viewCell dateWithSessions model timeDelimiters numColumns timeDelimiter index co
-            sessionStarting
-                |> Maybe.map .trackId
-                |> Maybe.withDefault 0
+            case (Maybe.map .trackId sessionStarting) of
```
duplication here. Also we can use [andThen](http://package.elm-lang.org/packages/elm-lang/core/latest/Maybe#andThen) to do this more succinctly

naazy replied:
Agree about duplication. I'm confused about how it would work with andThen. I've tried but it's not working :( I think the issue may be because we return 0 for the Nothing case but andThen returns Nothing for the Nothing case...

roryc89 commented:
Great stuff. Just a couple of suggestions.

naazy commented:
@roryc89 I've addressed your comments. Only thing is I have a question here: https://github.com/OxfordAbstracts/elm-program-builder/pull/61#discussion_r104948907

roryc89 commented:
all looking great! :)

nelsonic commented on tests/MainUpdateTests.elm:
```diff
@@ -130,7 +130,7 @@ all =
                         , chair = "test chair"
                         , location = "test location"
-                        , trackId = 1
+                        , trackId = (Just 1)
```
👍
